src/app/sessions/components/sessions-search/sessions-search.component.ts

```diff
@@ -24,39 +30,97 @@ export class SessionsSearchComponent implements OnInit {
     endDate;
     hearingParts$: Observable<HearingPart[]>;
     sessions$: Observable<SessionViewModel[]>;
+    rooms$: Observable<Room[]>;
+    judges$: Observable<Judge[]>;
     selectedSession: any;
     selectedHearingPartId;
+    filteredSessions$;
 
     constructor(private store: Store<fromHearingParts.State>) {
-        this.store.pipe(select(fromHearingParts.getHearingPartsEntities)).subscribe(data => {
-            this.hearingParts$ = Observable.of(data ? Object.values(data) : []);
-        });
+        this.hearingParts$ = this.store.pipe(select(fromHearingParts.getHearingPartsEntities),
+            map(this.asArray)) as Observable<HearingPart[]>;
+        this.rooms$ = this.store.pipe(select(fromSessions.getRooms), map(this.asArray)) as Observable<Room[]>;
+        this.judges$ = this.store.pipe(select(fromJudges.getJudges), map(this.asArray)) as Observable<Judge[]>;
+
         this.sessions$ = this.store.pipe(select(fromSessions.getFullSessions));
         this.startDate = moment().toDate();
         this.endDate = moment().add(5, 'years').toDate();
         this.selectedHearingPartId = '';
         this.selectedSession = {} ;
+        this.filteredSessions$ = this.sessions$;
     }
 
     ngOnInit() {
         this.store.dispatch(new SearchForDates({startDate: this.startDate, endDate: this.endDate}));
         this.store.dispatch(new fromHearingPartsActions.Search());
+        this.store.dispatch(new RoomActions.Get());
+        this.store.dispatch(new JudgeActions.Get());
     }
 
-    getSessions(startDate, endDate) {
-        this.store.dispatch(new SearchForDates({startDate: startDate, endDate: endDate}));
+    selectHearingPart(id: string) {
+        this.selectedHearingPartId = id;
     }
 
-    deassign() {
-        this.store.dispatch(new AssignToSession({
-            hearingPartId: this.selectedHearingPartId,
-            sessionId: null,
-            start: null
-        }));
+    filter(filters: SessionFilter) {
+        if (filters.startDate !== this.startDate) {
+            this.store.dispatch(new SearchForDates({startDate: filters.startDate, endDate: filters.endDate}));
+            this.startDate = filters.startDate;
+            this.endDate = filters.endDate;
+        }
+
+        this.sessions$.subscribe(data => {
+            this.filteredSessions$ = Observable.of(
+                data.filter(s => this.filterByProperty(s.person, filters.judges))
+                .filter(s => this.filterByProperty(s.room, filters.rooms))
+                .filter(s => this.filterByCaseType(s, filters))
+                .filter(s => this.filterByUtilization(s, filters.utilization))
+            )
+        })
     }
 
-    selectHearingPart(id: string) {
-        this.selectedHearingPartId = id;
+    private filterByCaseType(s, filters: SessionFilter) {
+        return filters.caseTypes.length === 0 ? true : filters.caseTypes.includes(s.caseType);
+    }
+
+    filterByUtilization(session: SessionViewModel, filters) {
```

Comment:
When we get to filtering on calendar view we might want to extract those filtering methods into separate service. For now it's fine as this story is not a priority right now.